File: src/main/js/eweb/src/app/ehome/addemployee/employeeDetailsRequestDto.ts
import {EmployeeFamilyDetailsRequestDto} from './employeeFamilyDetailsRequestDto';
import {EmployeePassportDetailRequestDto} from './employeePassportDetailRequestDto';
import {EmployeeWorkPermitDetailsRequestDto} from './employeeWorkPermitDetailsRequestDto';

export class EmployeeDetailsRequestDto{
addressLine1: string;
addressLine2: string;
city: string;
contactNumber:string;
country:string;
dateOfBirth:string;
department:string;
emailId:string;
firstName:string;
id: string;
jobRole: string;
lastName: string;
reportingPerson: string;
sex: string;
state: string;
workLocation: string;
familyDetails: EmployeeFamilyDetailsRequestDto[];
passportDetails: EmployeePassportDetailRequestDto[];
workPermitDetails: EmployeeWorkPermitDetailsRequestDto[];
constructor(){}
init(addressLine1: string,addressLine2: string,
city: string,
contactNumber:string,
country:string,
dateOfBirth:string,
department:string,
emailId:string,
firstName:string,
id: string,
jobRole: string,
lastName: string,
reportingPerson: string,
sex: string,
state: string,
workLocation: string,
workPermitDetails: EmployeeWorkPermitDetailsRequestDto[],
familyDetails: EmployeeFamilyDetailsRequestDto[],
passportDetails: EmployeePassportDetailRequestDto[]){
 this.addressLine1 = addressLine1;
 this.addressLine2 = addressLine2;
 this.city = city;
 this.contactNumber = contactNumber;
 this.country = country;
 this.dateOfBirth = dateOfBirth;
this.department = department;
this.emailId = emailId;
this.firstName = firstName;
this.id = id;
this.jobRole = jobRole;
this.lastName = lastName;
this.reportingPerson = reportingPerson;
this.sex = sex;
this.state = state;
this.workLocation = workLocation;
this.workPermitDetails = workPermitDetails;
this.familyDetails=familyDetails;
this.workPermitDetails=workPermitDetails;
}
}

File: src/main/java/com/employee/product/employeedetails/request/dto/EmployeeDetailsRequestDto.java
package com.employee.product.employeedetails.request.dto;

import java.sql.Date;
import java.util.List;


import lombok.Data;


@Data
public class EmployeeDetailsRequestDto {
	
	
	private Integer id;
	
	private String firstName;

	private String lastName;
	
	private String emailId;

	private String sex;
	
	private String addressLine1;

	private String addressLine2;
	
	private String city;
	
	private String state;
	
	private String country;
	
	private String contactNumber;
	
	private Date dateOfBirth;
	
	private String department;
	
	private String jobRole;
	
	private String reportingPerson;
	
	private String workLocation;
	
    private List<EmployeeWorkPermitDetailsRequestDto> workPermitDetails;
    
    private List<EmployeePassportDetailsRequestDto> passportDetails;
    
    private List<EmployeeFamilyDetailsRequestDto> familyDetails;
    
    private List<EmployeePaySlipDetailsRequestDto> paySlipDetails;

}
